pkg/serverless/serverless_test.go

```diff
@@ -30,14 +31,14 @@ func TestHandleInvocationShouldSetExtraTags(t *testing.T) {
 	d := daemon.StartDaemon("http://localhost:8124")
 	defer d.Stop()
 
-	d.WaitForDaemon()
+	// force daemon not to wait for flush at end of handleInvocation
+	d.SetFlushStrategy(flush.NewPeriodically(time.Second))
+	d.UseAdaptiveFlush(false)
 
-	d.TellDaemonRuntimeStarted()
+	// deadline = current time + 5s
+	deadlineMs := (time.Now().UnixNano())/1000000 + 5000
 
-	//deadline = current time + 20 ms
-	deadlineMs := (time.Now().UnixNano())/1000000 + 20
```

Comment:
We've seen two flaky test failures of this test in the last several weeks. I suspect they were due to this timeout being hit.

Therefore, I've increased the timeout enough so it won't be an issue in the future. Then, setting the flush strategy to not be "at the end" of each invocation, the daemon will not wait around for the data to flush. This means a faster test.